### PR TITLE
PP 284 implementation of outage messages

### PR DESCRIPTION
**Pull request by ValentinaIancu-Gini**
In this PR, we are handling separately some specific errors:
- 500 - outage error
- 503 - maintenance error
- 501...599 - server error message
- modified title and content for the generic error screen that will appear in case of error status code 499 and is also used as a fallback when an unknown error is received.
- fixed no internet connection error appearance
- code cleanup

### CODE REVIEW

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift [hunk 1]
```diff
@@ -16,7 +16,7 @@ import GiniBankAPILibrary
  - serverError: Error returned by the server.
  - authentication: Error related to authentication.
  - unexpected: Unexpected error that is not covered by the other cases.
- - importError: Error related to importing documents.
+ - maintenance: Error indicating that the system is under maintenance.
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift:19]:
```suggestion
 - maintenance: Error returned when the system is under maintenance.
```

**Review by zladzeyka — APPROVED** [review]
Great job! @ValentinaIancu-Gini 
🤟🏼

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift (hunk 1 above)
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift:19]:
```suggestion
 - maintenance: Error indicates that the system is under maintenance.
```

**Comment by ValentinaIancu-Gini** [comment]
@zladzeyka please check again. I fixed the test that was crashing. :facepalm:

**Review by zladzeyka — COMMENTED** [review]
We agreed that we'll try to refactor SessionManager/handleError. 🍀

**Comment by ValentinaIancu-Gini** [comment]
@zladzeyka I did the refactoring, fixed one test, and added one more. When you have time please check it.

File: BankAPILibrary/GiniBankAPILibrary/Sources/GiniBankAPILibrary/Documents/Core/SessionManager.swift [hunk 2]
```diff
@@ -331,11 +310,33 @@ private extension SessionManager {
             completion(.failure(.notAcceptable(response: response, data: data)))
         case 429:
             completion(.failure(.tooManyRequests(response: response, data: data)))
+        case 503:
+            completion(.failure(.maintenance))
+        case 500:
+            completion(.failure(.outage))
+        case 501, 502,504...599:
```
Comment (zladzeyka) [inline BankAPILibrary/GiniBankAPILibrary/Sources/GiniBankAPILibrary/Documents/Core/SessionManager.swift:317]:
```suggestion
        case 501, 502, 504...599:
```

**Review by zladzeyka — APPROVED** [review]
Looks good @ValentinaIancu-Gini, 
please find the tiny suggestion about a missing whitespace

**Pull request merged**